### ISSUE TITLE
Resolve inconsistency between SMART and OAuth regarding scopes for token exchange

### ISSUE THREAD

**Issue by arscan** [fhir-crucible/crucible_smart_app#68]
SMART says that the scope property is `required` in the body of the token exchange response.  Search for `Scope of access authorized. Note that this can be different from the scopes requested by the app` on http://www.hl7.org/fhir/smart-app-launch/

OAuth2 says that the scope property is optional if unchanged.  See https://tools.ietf.org/html/rfc6749#section-5.1

The application currently asserts that scope is included in the response body, as defined in the SMART on FHIR.  Since the failure occurs before the end of the test, we don't consider the test to have worked, as we don't store the bearer token.  One solution would be to fail that test, but to save the bearer token, therefore allowing the argo tests to run.

This may have been addressed during HL7 balloting.

**Comment by arscan** [comment]
This was addressed during balloting.  The resolution was to leave the additional SMART requirement in.

```“Jason Walonoski, MITRE, on behalf of Dylan Mahalingam, Mahalingam, Dylan <kmahalingam@mitre.org>
Comment: Some outside sources suggest that if a server provides access to exactly 
what is requested through scopes, the server does not need to return the scopes 
provided in its response, since they are identical to those requested. Is this true? 
If so, perhaps add a line to mention this.
Summary: Question about scopes in response from authorization server.”

“While the OAuth spec allows a server to omit scopes, our specification can be tighter
 and prevent it. No change required.”```

However, since we do get the token back, we can continue with the later tests.  Let's save the bearer token before doing this test, but still fail it.

**Comment by kjmahalingam** [comment]
Resolved in #72.
